docs: sync schema (#613)
bytebase/bytebase.com: content/docs/change-database/synchronize-schema.md

---
title: Synchronize Schema
---

<hint-block type="warning">

This feature is in beta and only available for MySQL.

</hint-block>

> Synchronize schema can identify the inconsistent database schema and get the DDL statements through the structure comparison function. The user can synchronize one database schema to another database.

Reviewer: ```suggestion
Bytebase can synchronize one database schema to another database. To synchronize the schema, Bytebase calculates the schema diff between the two databases and generates the required DDL statements (e.g. ALTER TABLE).
```

> Bytebase can synchronize one database schema to another database. To synchronize the schema, Bytebase calculates the schema diff between the two databases and generates the required DDL statements (e.g. ALTER TABLE).

Reviewer 2: ```suggestion
Bytebase supports synchronizing a specified schema version of one database to another. To provide a better user experience, Bytebase displays the schema differences between the two selected databases and generates the suggested DDL statements (e.g. ALTER TABLE).
```

## How to use

> 

Reviewer 2: It would be great to tell users how to launch the dialog below.

![sync-schema-dialog](/static/docs/change-database/synchronize-schema/sync-schema-dialog.webp)

> 1. Select project

Reviewer 2: We recommend replacing "1." with "Step 1 -".
![CleanShot 2022-11-14 at 16 11 02@2x](https://user-images.githubusercontent.com/104962472/201608238-c34baa55-4194-4c57-9dd7-f4ccfe63be54.png)

> 2. Select schema

Reviewer: ```suggestion
2. Select schema version
```

>    The schema should be one of the database's migration history. The default is the latest.

Reviewer: ```suggestion
   The schema version is selected from the database's migration history.
```

   <hint-block type="info">

>    In the Enterprise Plan, you can set a special schema version from migration history list.

Reviewer: ```suggestion
   In the Enterprise Plan, you can select an arbitrary schema version from the migration history. In other plans, you can only select the latest schema version.
```

   </hint-block>

3. Apply to database

>    Select the database that will be synchronized with the previously selected schema.

Reviewer: ```suggestion
   Select the database that will be synchronized with the chosen schema version in the last step.
```

> 4. The generated DDL statement

Reviewer: ```suggestion
4. Generated DDL statements
```

> 4. Generated DDL statements

Reviewer 2: I think "Generated DDL statements" is a result of Step 3. So I suggest merging Step 4 into Step 3.

>    Bytebase will show a code diff between two database schema and generate the DDL statement. You can further edit the generated DDL statement.

Reviewer: ```suggestion
   Bytebase will show a schema diff between two databases and generate the DDL statements. You can further edit the generated content.
```

5. Preview issue

>    With the generated DDL statement, you can preview and create an "Alter Schema" issue for the selected database.

Reviewer: ```suggestion
   With the generated DDL statements, you can preview and create an "Alter Schema" issue. This kicks off the [change workflow](/docs/change-database/change-workflow) for the applied database.
```

> After you have selected a schema and database, Bytebase will show a schema diff between two databases and generate the DDL statements. And you can further edit the generated content.

Reviewer: ```suggestion
After you have selected a schema and database, Bytebase will show a schema diff between the two databases and generate the DDL statements. And you can further edit the generated content.
```
